src/System.Collections.NonGeneric/tests/CaseInsensitiveComparerTests.cs

```diff
@@ -119,31 +120,40 @@ public static void Ctor_CultureInfo_NullCulture_ThrowsArgumentNullException()
         [InlineData("hello", "hello", 0)]
         [InlineData("HELLO", "HELLO", 0)]
         [InlineData("hello", "goodbye", 1)]
-        [InlineData("hello", null, 1)]
-        [InlineData(null, "hello", -1)]
+        [InlineData("hello", "null", 1)]
+        [InlineData("null", "hello", -1)]
         [InlineData("file", "FILE", 0)] // Turkey's comparing system is ignored as this is invariant
         [InlineData(5, 5, 0)]
         [InlineData(10, 5, 1)]
         [InlineData(5, 10, -1)]
-        [InlineData(5, null, 1)]
-        [InlineData(null, 5, -1)]
-        [InlineData(null, null, 0)]
-        public static void DefaultInvariant_Compare(object a, object b, int expected)
+        [InlineData(5, "null", 1)]
+        [InlineData("null", 5, -1)]
+        [InlineData("null", "null", 0)]
+        public void DefaultInvariant_Compare(object a, object b, int expected)
         {
-            CultureInfo culture1 = CultureInfo.CurrentCulture;
-            CultureInfo culture2 = CultureInfo.CurrentUICulture;
-
-            try
+            RemoteInvoke((ra, rb, rexpected) =>
             {
-                var cultureNames = new string[]
+                Func<string, object> convert = (string o) =>
                 {
-                    "cs-CZ","da-DK","de-DE","el-GR","en-US",
-                    "es-ES","fi-FI","fr-FR","hu-HU","it-IT",
-                    "ja-JP","ko-KR","nb-NO","nl-NL","pl-PL",
-                    "pt-BR","pt-PT","ru-RU","sv-SE","tr-TR",
-                    "zh-CN","zh-HK","zh-TW"
+                    if (Int32.TryParse(o, out int ret))
+                        return ret;
+
+                    return (o == "null") ? null : o;
                 };
 
+                var ra_val = convert(ra);
+                var rb_val = convert(rb);
+                var rexpected_val = convert(rexpected);
+
```

Comment:
@danmosemsft, I took a quick pass at it here:
https://github.com/stephentoub/corefx/commit/7e7b200737a5b8d96bf8cb6345abaeee796b21fe

I'll let you push it across the finish line if you're keen on it.  A few things:
1) Most RemoteInvoke usage isn't affected, as they're parameterless.  But I actually think the ones that are affected were better off without the change.
2) A few of the tests I commented out as they relied on actually passing data via the command-line, and this change passes the data via an input file.
3) It looks like there's now a separate RemoteInvoke implementation for uap, but it's not clear to me that it's been kept up-to-date (e.g. it's missing your exception change), and I can't find the actual host app anywhere in the repo, so while I paid lip service to fixing that up, I haven't verified it at all and doubt it actually works.